### PR TITLE
Add cf health-check configuration to deployment

### PR DESCRIPTION
**Pull request by krzysiekb**
Add possibility to configure CF health-check in deployment.

```
healthCheck {
   type = "http"
   endpoint = "/health"
   timeout = 60
   invocationTimeout = 60
}
```

Basic validation was added for health-check parameters, when `healthCheck` stanza exists in HCL file:
- `type` is required
- `endpoint` is required for type `http`
- `timeout` is optional, when configured must be between 0 and 180 seconds
- `invocationTimeout` is optional, when configured must be between 0 and 180 seconds

For details about CF health-check, please consult: [How to use CF Health-Checks](https://docs.cloudfoundry.org/devguide/deploy-apps/healthchecks.html)

### CODE REVIEW

File: platform/platform.go [hunk 1]
```diff
@@ -407,18 +426,75 @@ func (p *Platform) configureQuota(state DeploymentState) error {
...
+		}
+
+		step.Done()
+	}
+	return nil
+}
+
+func needToConfigureHealthCheck(state *DeploymentState) bool {
+	return state.healthCheckParams != nil && (state.healthCheckParams.Endpoint != "" ||
+		state.healthCheckParams.Type != "" || state.healthCheckParams.InvocationTimeout != 0 ||
```
Comment (denysvitali) [inline platform/platform.go:475]: Are you sure it's needed? I would only check if `state.healthCheckParms` is not `nil`.

**Review by denysvitali — CHANGES_REQUESTED** [review]

File: platform/platform.go (hunk 1 above)
Comment (krzysiekb) [reply platform/platform.go:475]: I guess we can limit the check to only top level field. There's validation later, that would catch other problems anyway.

Comment (denysvitali) [reply platform/platform.go:475]: Awesome! I'll then simplify this part and remove entirely this function :)

Thanks!

Comment (krzysiekb) [reply platform/platform.go]: Did it already :-)

**Pull request merged**